### PR TITLE
[skip-ci]: feat(add): Backup/restore experiments for CSI volumes

### PR DESCRIPTION
**Pull request by shashank855**
Signed-off-by: <ranjanshashank855@gmail.com>

Adds following experiments CSI based backup/restore experiments into the piepeline:
- Backup/restore 
- Remote restore in different namespace.
- Backup/Restore using different S3 profile.

# PR fixes following issue:
- https://github.com/openebs/e2e-tests/issues/406

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -505,6 +505,24 @@ HODS-CSI-PROVISIONER:
   dependencies:
     - SUTP-konvoy-cluster 
 
+XYNP-Remote-restore-diff-ns:
```
Comment (w3aman) [inline .gitlab-ci.yml:508]: Backup-restore-different-profile ***

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+#echo "Creating configmap in litmus namespace for GCP token"
+#kubectl create cm gcp-key -n litmus --from-file /var/sdk/key.json
+
+cp apps/percona/deployers/run_litmus_test.yml percona_backup_restore_csi.yml
+
+sed -i -e 's/app: percona-deployment/app: deploy-percona-backup-restore-csi/g' \
+-e 's/value: openebs-standard/value: openebs-cstor-csi/g' \
+-e 's/value: app-percona-ns/value: percona-backup-csi/g' percona_backup_restore_csi.yml
+
```
Comment (nsathyaseelan) [inline openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi:53]: @shashank855  can you append with run_id for the test_name.

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+case_id=HNOV
+
+time="date"
+current_time=$(eval $time)
+
+present_dir=$(pwd)
+echo $present_dir
+
+bash openebs-konvoy-e2e/utils/e2e-cr jobname:backup-restore-cstor-csi jobphase:Waiting
+bash openebs-konvoy-e2e/utils/e2e-cr jobname:backup-restore-cstor-csi jobphase:Running init_time:"$current_time" jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id" openebs_version:"$releaseTag"
```
Comment (nsathyaseelan) [inline openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi:28]: @shashank855  where is the Waiting `e2e-cr` for backup restor using diss profile and backup restore diff ns

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+cd litmus
+cp experiments/functional/backup_and_restore/run_litmus_test.yml run_backuprestore_test_cstor_csi.yml
+
+sed -i -e 's/value: app-percona-ns/value: percona-backup-csi/g' \
+-e 's/app: velero-backup-restore/app: velero-backup-restore-cstor-csi/g' \
+-e 's/value: "minio"/value: "GCP"/g' \
+-e '/name: DEPROVISION_VELERO/{n;s/.*/            value: "true"/}' run_backuprestore_test_cstor_csi.yml
+
+sed -i "s/v1.1.0/$velero_version/g" run_backuprestore_test_cstor_csi.yml
```
Comment (nsathyaseelan) [inline openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi:83]: @shashank855  in litmusbook i think the value as changed into v1.3.0 as a default env?

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/XYNP-Remote-Restore-Diff-ns/remote-restore-diff-ns [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+cat percona_backup_restore_csi_diff_ns.yml
+
+bash ../openebs-konvoy-e2e/utils/litmus_job_runner label='app:deploy-percona-backup-restore-csi-diff-ns' job=percona_backup_restore_csi_diff_ns.yml
+cd ..
+bash openebs-konvoy-e2e/utils/dump_cluster_state;
+bash openebs-konvoy-e2e/utils/event_updater jobname:backup-restore-cstor-csi-diff-ns $test_name jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
+
+if [ "$?" != "0" ]; then
+python3 openebs-konvoy-e2e/utils/result/result_update.py $job_id XYNP 3-functional "Create backup and restore and verify if the application is not impacted" Fail $pipeline_id "$current_time" $commit_id $gittoken
```
Comment (nsathyaseelan) [inline openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/XYNP-Remote-Restore-Diff-ns/remote-restore-diff-ns:63]: @shashank855  can you provide the exact test description for each test.

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi (hunk 4 above)
Comment (shashank855) [reply openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi:83]: No its not.

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi (hunk 3 above)
Comment (shashank855) [reply openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi:28]: Look at script remote-restore-diif-ns line-no: 27 & backup-restore-diff-profile line-no: 27

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/XYNP-Remote-Restore-Diff-ns/remote-restore-diff-ns (hunk 5 above)
Comment (shashank855) [reply openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/XYNP-Remote-Restore-Diff-ns/remote-restore-diff-ns:63]: I have made changes for remote restore in diff ns and backup/restore using different s3 profile.

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi (hunk 3 above)
Comment (nsathyaseelan) [reply openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi:28]: For pooling logic we no need to create e2e-cr in the very first dependent job? @shashank855

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -505,6 +505,24 @@ HODS-CSI-PROVISIONER:
   dependencies:
     - SUTP-konvoy-cluster 
 
+XYNP-Backup-restore-diff-profile:
```
Comment (nsathyaseelan) [inline .gitlab-ci.yml:508]: @shashank855  test caseid is 0TQI

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/XYNP-Remote-Restore-Diff-ns/remote-restore-diff-ns [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+################
+# LitmusBook 2 #
+################
+
+echo "*******Performing Backup and Restore on Target**********"
+
+test_name=$(bash openebs-konvoy-e2e/utils/generate_test_name testcase=velero-backup-restore metadata="")
+echo $test_name
+
```
Comment (nsathyaseelan) [inline openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/XYNP-Remote-Restore-Diff-ns/remote-restore-diff-ns:80]: @shashank855  here also append with run_id? Wherever using the same testcase name in alll the places can you append

File: openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi (hunk 3 above)
Comment (shashank855) [reply openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi:28]: @nsathyaseelan If you look at:
 https://github.com/mayadata-io/e2e-openshift/blob/a696ca677162f9943fc419bb48bbe0de5d690360/Openshift-EE/pipelines/OpenEBS-base/stages/3-functional/3PZ0-Backup-Restore-Cstor/backup-restore-cstor#L27
It works fine in e2e-openshift pipeline.

Comment (shashank855) [reply openebs-konvoy-e2e/pipelines/OpenEBS-base/stages/7-csi-functional/HNOV-Backup-Restore-Cstor/backup-restore-cstor-csi:28]: IMO it will not break.

**Pull request merged**